### ISSUE TITLE
Nvidia proprietary driver installation

### ISSUE THREAD

**Issue by kaan35** [archlinux/archinstall#726]
Hello 

I tried to install my system via this installer but i can't install nvidia proprietary driver with this installer.

I think there is an option "Nvidia (proprietary)" but it still installing open source nvidia driver.
https://github.com/archlinux/archinstall/blob/0071a069080732047e11309134869f3ab40c642c/archinstall/lib/hardware.py

How can install proprietary drivers ?
Instead of gfx_driver option adding nvidia package names to packages option like that :
```"packages": ["bash-completion","linux-headers","nvidia","nvidia-settings","opencl-nvidia"]```

My config file :

```
{
  "bootloader": "systemd-bootctl",
  "audio": "pulseaudio",
  "filesystem": "ext4",
  "gfx_driver": "Nvidia (proprietary)",
  "harddrive": {
   "path": "/dev/sda"
  },
  "hostname": "pc",
  "kernels": ["linux"],
  "keyboard-language": "trq",
  "mirror-region": {},
  "nic": {
    "NetworkManager": true
  },
  "profile": "kde",
  "ntp": true,
  "packages": ["bash-completion", "linux-headers"],
  "sys-encoding": "utf-8",
  "timezone": "UTC",
  "!root-password": "****",
  "superusers": {
    "kaan": {
      "!password": "****"
      }
  }
}
```

Thank you.

**Comment by OhKannaDuh** [comment]
It should be `"gfx_driver": "Nvidia",` I think

**Comment by dylanmtaylor** [comment]
> It should be `"gfx_driver": "Nvidia",` I think

I think what he has is correct. If it's not working, this is a regression -- pretty sure it used to work

**Comment by OhKannaDuh** [comment]
You could be right, I was basing this off the values listed here: https://github.com/archlinux/archinstall/blob/master/docs/installing/guided.rst

**Comment by dylanmtaylor** [comment]
> You could be right, I was basing this off the values listed here: https://github.com/archlinux/archinstall/blob/master/docs/installing/guided.rst

We'll have to go through that and update some values, as the guided partitioner's options have changed a bit, and the config uses the same strings.

**Comment by Torxed** [comment]
Most likely it's the `generic_select` returning the key instead of the value from these options: https://github.com/archlinux/archinstall/blob/0071a069080732047e11309134869f3ab40c642c/archinstall/lib/hardware.py#L56

**Comment by kaan35** [comment]
> It should be `"gfx_driver": "Nvidia",` I think

Yes i tried it first but it is installed open source drivers.

**Comment by Torxed** [comment]
So assuming either `archinstall --gfx_driver='Nvidia (proprietary)'` is set or the config contains:
```json
{
    "gfx_driver": "Nvidia (proprietary)"
}
```
It should work because guided will use the above parameter: https://github.com/archlinux/archinstall/blob/7d991ecb9f87f863e1e78ce7e2d06c4d2f9568db/examples/guided.py#L49-L50

And perform a lookup in `AVAILABLE_GFX_DRIVERS`: https://github.com/archlinux/archinstall/blob/7d991ecb9f87f863e1e78ce7e2d06c4d2f9568db/archinstall/lib/hardware.py#L23-L58

@kaan35 How have you verified that it's *not* the `nvidia` package being installed? Because everything looks fine.

**Comment by kaan35** [comment]
@Torxed
When i run this command : `lspci -k | grep -A 2 -E "(VGA|3D)"` it returns nouveau is using.
So I installed nvidia packages after installation.
I run this command on my konsole :` pacman -S nvidia nvidia-settings  nvidia-utils`

Now  `lspci -k | grep -A 2 -E "(VGA|3D)"` returns 
```01:00.0 VGA compatible controller: NVIDIA Corporation GM107 [GeForce GTX 750 Ti] (rev a2)
        Subsystem: ASUSTeK Computer Inc. Device 84bb
        Kernel driver in use: nvidia
```

**Comment by Torxed** [comment]
Do you have the `/var/log/archinstall/install.log` still? Maybe that could shed some light on what happned.
Because you are correct, `nouveau` should not be installed, it should be `nvidia`.

**Comment by kaan35** [comment]
@Torxed 

Output : 

`````
Disk states before installing: {'blockdevices': [{'name': 'loop0', 'fstype': 'squashfs', 'fsver': '4.0', 'label': None, 'uuid': None, 'fsavail': '0', 'fsuse%': '100%', 'mountpoints': ['/run/archiso/airootfs'], 'type': 'loop', 'size': '673.1M'}, {'name': 'sda', 'fstype': None, 'fsver': None, 'label': None, 'uuid': None, 'fsavail': None, 'fsuse%': None, 'mountpoints': [None], 'type': 'disk', 'size': '232.9G'}, {'name': 'sdb', 'fstype': None, 'fsver': None, 'label': None, 'uuid': None, 'fsavail': None, 'fsuse%': None, 'mountpoints': [None], 'type': 'disk', 'size': '57.8G', 'children': [{'name': 'sdb1', 'fstype': 'vfat', 'fsver': 'FAT32', 'label': 'ARCH_202110', 'uuid': '12F0-0C3F', 'fsavail': '57.1G', 'fsuse%': '1%', 'mountpoints': ['/mnt/usb', '/run/archiso/bootmnt'], 'type': 'part', 'size': '57.8G'}]}]}
-- Guided template chosen (with below config) --
{
    "!root-password": "******",
    "audio": "pulseaudio",
    "bootloader": "systemd-bootctl",
    "filesystem": "ext4",
    "gfx_driver": "Nvidia (proprietary)",
    "harddrive": {
        "model": "Samsung SSD 750 EVO 250GB",
        "path": "/dev/sda",
        "size": "232.9G"
    },
    "hostname": "pc",
    "kernels": [
        "linux"
    ],
    "keyboard-language": "trq",
    "mirror-region": {},
    "nic": {
        "NetworkManager": true
    },
    "ntp": true,
    "packages": [
        "bash-completion",
        "linux-headers"
    ],
    "profile": {
        "path": "/usr/lib/python3.9/site-packages/archinstall/profiles/kde.py"
    },
    "script": "guided",
    "silent": false,
    "superusers": {
        "kaan": {
            "!password": "******"
        }
    },
    "sys-encoding": "utf-8",
    "sys-language": "en_US",
    "timezone": "UTC"
}
Wiping BlockDevice(/dev/sda) by using partition format 1
Using and formatting the entire BlockDevice(/dev/sda).
Adding partition to BlockDevice(/dev/sda)
Adding partition to BlockDevice(/dev/sda)
Set the root partition Partition(path=/dev/sda2, size=232.4G, fs=ext4) to use filesystem ext4.
Formatting /dev/sda1 -> vfat
Formatting /dev/sda2 -> ext4
Mounting Partition(path=/dev/sda2, size=232.4G, fs=ext4, rel_mountpoint=/) to /mnt
Mounting Partition(path=/dev/sda1, size=512M, fs=vfat, rel_mountpoint=/boot) to /mnt/boot
Waiting for automatic mirror selection (reflector) to complete.
Installing packages: ['base', 'base-devel', 'linux-firmware', 'linux', 'intel-ucode']
Adding bootloader systemd-bootctl to Partition(path=/dev/sda1, size=-1, fs=vfat, mounted=/mnt/boot)
Installing packages: ('efibootmgr',)
Identifying root partition by PART-UUID on Partition(path=/dev/sda2, size=-1, fs=ext4, mounted=/mnt), looking for 'a51a81fb-36f5-444a-a8ec-60c8ceac38a4'.
Installing packages: ('networkmanager',)
Enabling service NetworkManager.service
This audio server will be used: pulseaudio
Installing packages: ('pulseaudio',)
Installing packages: ['bash-completion', 'linux-headers']
Installing network profile Profile(kde)
Installing network profile Profile(xorg)
Installing packages: ('xorg-server xorg-xinit',)
Installing packages: ['plasma-meta', 'konsole', 'kate', 'dolphin', 'sddm', 'plasma-wayland-session', 'egl-wayland']
Enabling service sddm
Creating user kaan
Setting password for kaan
Enabling sudo permissions for kaan.
Installing and activating NTP.
Installing packages: ('ntp',)
Enabling service ntpd
Setting password for root
Keyboard language for this installation is now set to: trq
For post-installation tips, see https://wiki.archlinux.org/index.php/Installation_guide#Post-installation
Updating /mnt/etc/fstab
Installation completed without any errors. You may now reboot.



```

**Comment by Torxed** [comment]
Very strange. It never installed *any* graphics driver *(`grep "Installing packages" /var/log/archinstall/install.log`)*.
I see that `Profile(xorg)` did get installed as expected. And I think I just found the issue:
We have a silent exception that most likely kicked in: https://github.com/archlinux/archinstall/blob/776823adfb2d9ea140b9d849ee9a1dd07991a94c/profiles/xorg.py#L49-L50

**Comment by Torxed** [comment]
I've submitted a patch to at least clarify what's going on.
I'm building and testing against it right now. See #731

**Comment by Torxed** [comment]
Funny enough, I think the linter just spotted the *real* issue: https://github.com/archlinux/archinstall/runs/4251813523?check_suite_focus=true

It's the use of "f-strings" which were inconsistent and escaped poorly. I wonder if this ever worked.

**Comment by kaan35** [comment]
@Torxed 
Thank you so much.
I should change any thing in my config.json file ?

**Comment by Torxed** [comment]
No need to change anything. This was a programmatic error on our end. Good job finding it and thank you for reporting it!
Should be fixed in master now : )

I've verified *(see last log output)*:
![screenshot](https://user-images.githubusercontent.com/861439/142424928-5d22c788-e25e-4747-8f96-d0fd57e94f79.jpg)

**Comment by kaan35** [comment]
Thank you so much 

How can use this version of installer ? 
I should wait next release or install from this master branch ?

**Comment by Torxed** [comment]
> Thank you so much
> 
> How can use this version of installer ? I should wait next release or install from this master branch ?

I will put up a release candidate tonight, sign it and maybe grazzolini have a few minutes to spare and wants to spend those putting up a new Arch artifact on the release. That way you can download the RC2 release ISO (and signature) or use `pacman -U artifact` to update the live ISO.

But if you wanna try it right away, you can use the unsupported ISO that github builds, it always contains the latest master changes: https://github.com/archlinux/archinstall/actions/runs/1476674656

*(Being built as I write this, usually takes 15min or so)*

**Comment by kaan35** [comment]
I see. I will try. Thank you.

**Comment by kaan35** [comment]
@Torxed 
I tried to install with latest version of installer. I can't add user while installing system with installer.
In version 2.2.0 asking user name and password to create but in master version i can't create user while using installer even if i added in config.json.

Thank you.

**Comment by kaan35** [comment]
I installed system and added user according to this config.json

```{
    "audio": "pulseaudio",
    "custom-commands": [
        "useradd -m kaan -G wheel; echo -e '123456\n123456' | passwd kaan"
    ],
    "desktop-environment":"kde",
    "filesystem": "ext4",
    "gfx_drivers": "VMware / Virtualbox (open-source)",
    "harddrives": ["/dev/sda"],
    "hostname": "pc",
    "kernels": ["linux"],
    "keyboard-layout": "trq",
    "nic": {"nic": "enp0s3"},
    "ntp": true,
    "packages": ["bash-completion","linux-headers"],
    "sys-encoding": "utf-8",
    "timezone": "Europe/Istanbul",
    "!root-password": "123456",
    "script":"guided",
    "silent":false,
    "swap":true
}
```

**Comment by Torxed** [comment]
`master` requires you to supply `--creds users.json` or an entry in the config like this:

```json
{
    "users": {
        "Anton": {"!password": "secret"}
    }
}
```

**Comment by kaan35** [comment]
I see. I didn't know that sorry. Thank you for reply.

**Comment by Torxed** [comment]
No worries :) It's still a relatively new thing about to be documented ^^

**Comment by kaan35** [comment]
Hello

I am trying to install on my virtual machine but installer fails at first disk format step.

I uninstalled archinstall included in iso and i cloned this repo then i installed required python packages to run this script.

Error message 
![image](https://user-images.githubusercontent.com/25592633/149911917-639a8a36-9520-43a9-871c-fa550410d97e.png)


I tried with --config and --creds and without json files, both of them gives same error message.

This is output of install.log file
![image](https://user-images.githubusercontent.com/25592633/149912402-66861948-0af9-4423-b59b-0fc7cf229e5d.png)

![image](https://user-images.githubusercontent.com/25592633/149915558-9d371633-518a-41d7-a044-cec6aa69752e.png)

![image](https://user-images.githubusercontent.com/25592633/149915607-e186e48b-2b3d-46a0-b285-def5999827c7.png)


Thank you :)
